code/modules/mob/living/carbon/human/species_types/shadowpeople.dm

```diff
@@ -21,3 +22,60 @@
 			H.take_overall_damage(1,1)
 		else if (light_amount < 0.2) //heal in the dark
 			H.heal_overall_damage(1,1)
+
+
+/datum/species/shadow/nightmare
+	name = "Nightmare"
+	id = "nightmare"
+	burnmod = 1.5
+	blacklisted = TRUE
+	species_traits = list(NOBREATH,RESISTCOLD,RESISTPRESSURE,NOGUNS,NOBLOOD,RADIMMUNE,VIRUSIMMUNE,PIERCEIMMUNE,NODISMEMBER,NO_UNDERWEAR)
+	mutanteyes = /obj/item/organ/eyes/night_vision/nightmare
+
+/datum/species/shadow/nightmare/on_species_gain(mob/living/carbon/C, datum/species/old_species)
+	. = ..()
+	C.AddSpell(/obj/effect/proc_holder/spell/targeted/shadowwalk)
+	var/obj/item/light_eater/blade = new
+	C.put_in_hands(blade)
+
+
+/datum/species/shadow/nightmare/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
+	var/light_amount = 0
+	if(isturf(H.loc))
+		var/turf/T = H.loc
+		light_amount = T.get_lumcount()
+		if (light_amount < 0.2)
+			H.visible_message("<span class='danger'>[H] dances in the shadows, evading [P]!</span>")
+			playsound(T, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, 1)
+			return -1
+	return 0

```

Comment:
I'd replace this and `return -1` with just `return`, unless `return -1` is explicitly checked somehow

Reply:
It is

Reply:
Oh, neat. Okey doke.